### PR TITLE
model, ci: build image using published versions of snaps

### PR DESCRIPTION
**Pull request by jhenstridge**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,40 +1,17 @@
...
-	cat extra-gadget.yaml >> pc-desktop/meta/gadget.yaml
-	cp cloud.conf pc-desktop/cloud.conf
-	cp setup.sh pc-desktop/setup.sh
-	snap pack --filename=$@ pc-desktop
-
-pc.img: gdm-spike-model.model pc-desktop.snap $(EXTRA_SNAPS)
+pc.img: gdm-spike-model.model
 	rm -rf img/
 	ubuntu-image snap --output-dir img --image-size 8G \
 	  --snap pc-desktop.snap $(foreach snap,$(ALL_SNAPS),--snap $(snap)) $<
```
Comment (kenvandine) [inline Makefile:10]: @jhenstridge looks like you missed dropping pc-desktop.snap here

**Review by kenvandine — APPROVED** [review]
LGTM

**Comment by jhenstridge** [comment]
Reverting to the UC20 based gadget snap results in the same problems I was having with the UC22 based gadget, so that's not an issue. I end up with a system with no snapd available.

The `state.json` file shows that it's failing on the task "Connect firefox:desktop to ubuntu-desktop-session:desktop" with:

```
2022-12-06T05:54:12Z ERROR connection denied by slot rule of interface "desktop"
```

Then everything else gets undone leaving the system in a broken state. This isn't happening for the main builds because the unpublished ubuntu-desktop-session bypasses the connection checks in the base declaration. The base declaration is probably correct in this instance, since we wouldn't want random snaps providing desktop slots. So this means we need a snap declaration for ubuntu-desktop-session to allow it (probably along with the wayland interface too).

**Comment by jhenstridge** [comment]
The snap-declaration for ubuntu-desktop-session is now sufficient to correctly seed this image. The `allow-auto-connect` rule for the wayland slot is buggy though, that interface doesn't auto-connect for new snaps like the desktop interface does. Hopefully that might get fixed later today.

With this, the only unpublished snap in the image is snapd itself. We also end up missing out on some of the work the cloud-init script was doing. Namely:

```sh
if [ -d /var/snap/network-manager/current/ ]; then
    mkdir -p /var/snap/network-manager/current/conf.d
    cat > /var/snap/network-manager/current/conf.d/disable-polkit.conf <<EOF
[main]
auth-polkit=false
EOF
fi

#initalize lxd with zfs storage backend for sudo support
/snap/bin/lxd init --storage-backend zfs --auto
```

All the interface connection now happens automatically as the snaps are installed.

**Comment by kenvandine** [comment]
@jhenstridge shouldn't we keep cloud-init then?

**Comment by jhenstridge** [comment]
I've added the remaining bits of the cloud-init script to the published pc-desktop snap in https://github.com/canonical/pc-amd64-gadget-desktop/commit/4dd3a195498a5100864d1ca167ea5f75d818330a, so this now seems to have parity with the main branch.

**Pull request merged**